src/app/routes/cpsAsset/getInitialData/addRecommendationsBlock/index.js

```diff
@@ -0,0 +1,48 @@
+import path from 'ramda/src/path';
+import deepClone from 'ramda/src/clone';
+import splitAt from 'ramda/src/splitAt';
+import { getNthCpsParagraphIndex } from '../helpers';
+
+const insertRecommendationsBlock = (recommendationBlock, blocks) => {
+  // get the index of the 5th paragraph
+  const fifthParagraphIndex = getNthCpsParagraphIndex(blocks, 5);
+
+  if (!fifthParagraphIndex) {
+    return blocks;
+  }
+
+  // split blocks at the index of the 5th paragraph
+  const parts = splitAt(fifthParagraphIndex + 1, blocks);
+
+  // reconstruct blocks
+  return [...parts[0], { ...recommendationBlock }, ...parts[1]];
+};
+
+const addRecommendationsBlock = originalJson => {
+  const json = deepClone(originalJson);
+  const pageType = path(['metadata', 'type'], json);
+  const { allowAdvertising } = path(['metadata', 'options'], json);
+  const blocks = path(['content', 'model', 'blocks'], json);
+
+  if (pageType !== 'STY' || !blocks || !allowAdvertising) {
+    return json;
+  }
+
+  const { assetUri } = path(['metadata', 'locators'], json);
+
+  const block = {
+    type: 'wsoj',
+    model: {
```

Comment:
is there any disadvantaged to just composing the entire path for the camino endpoint here? would save composing it in the container.

Reply:
none actually, I'll go for it